scripts/casefold_db.py

```diff
@@ -18,16 +18,58 @@
 import os
 import sqlite3
 import sys
+import time
 from typing import Any, Dict, List, Tuple
 
+import attr
 import signedjson.sign
 
 from sydent.sydent import Sydent, parse_config_file
 from sydent.util import json_decoder
-from sydent.util.emailutils import sendEmail
+from sydent.util.emailutils import EmailSendException, sendEmail
 from sydent.util.hash import sha256_and_url_safe_base64
 from tests.utils import ResolvingMemoryReactorClock
 
+EMAIL_SUBJECT = "No action required: we have changed the way your Matrix account and email address are associated"
+
+# Maximum number of attempts to send an email.
+MAX_ATTEMPTS_FOR_EMAIL = 10
```

Comment:
If we're not removing the rows from the DB on failure, I'd be inclined to be less aggressive as the admin can just rerun the script later. I'd suggest something like 3? Or 5? 3 would mean a retry schedule 1s -> 2s and so a total of 3s waiting. 5 would mean 1s -> 2s -> 4s -> 8s  and so a total of 15s waiting.10 would be ~9mins.

Reply:
Yeah I wasn't entirely sure about 10, I think I'm going to go with 5 because 3 would be a bit tight when working with mailtrap (so might also be the case with other servers) iirc